db/migrate/20190605014649_add_purpose_addons.rb

```diff
@@ -0,0 +1,29 @@
+class AddPurposeAddons < ActiveRecord::Migration[5.2]
+  def change
+    create_table :katello_purpose_addons do |t|
+      t.string :name, null: false
+    end
+
+    create_table :katello_subscription_facet_purpose_addons do |t|
+      t.references :purpose_addon, index: { name: :katello_sub_facet_purpose_addons_paid }
+      t.references :subscription_facet, index: { name: :katello_sub_facet_purpose_addons_sfid }
+    end
+
+    add_foreign_key :katello_subscription_facet_purpose_addons, :katello_subscription_facets, column: :subscription_facet_id, name: :katello_sub_facet_purpose_addon_facet_id
+    add_foreign_key :katello_subscription_facet_purpose_addons, :katello_purpose_addons, column: :purpose_addon_id, name: :katello_sub_facet_purpose_addon_purpose_addon_id
+
+    Katello::Host::SubscriptionFacet.pluck(:id, :purpose_addons).each do |facet|
```

Comment:
I was most concerned if this migration would work properly or not - and it seems to work great. Here's what I did:

Set up some data:

```
irb(main):010:0> Katello::Host::SubscriptionFacet.where.not(purpose_addons: []).pluck(:id, :purpose_addons)
=> [[265, ["TestAddon1", "TestAddon2"]], [257, ["TestAddon1", "TestAddon2"]], [271, ["TestAddon3"]], [274, ["TestAddon1"]]]
```

Applied code change & migrated db, got these results:

```
irb(main):003:0> Katello::PurposeAddon.pluck(:name)
=> ["TestAddon1", "TestAddon2", "TestAddon3"]

Katello::Host::SubscriptionFacet.find(265).purpose_addons.pluck(:name)
=> ["TestAddon1", "TestAddon2"]

Katello::Host::SubscriptionFacet.find(257).purpose_addons.pluck(:name)
=> ["TestAddon1", "TestAddon2"]

Katello::Host::SubscriptionFacet.find(271).purpose_addons.pluck(:name)
=> ["TestAddon3"]

Katello::Host::SubscriptionFacet.find(274).purpose_addons.pluck(:name)
=> ["TestAddon1"]

Katello::SubscriptionFacetPurposeAddon.pluck(:subscription_facet_id, :purpose_addon_id)
=> [[265, 1], [265, 2], [257, 1], [257, 2], [271, 3], [274, 1]]
```

Reply:
Yes. This looks like the tables are set up the way we want. :+1: